pentaho/impl/src/main/java/pt/webdetails/cdf/dd/cache/impl/Cache.java

```diff
@@ -35,41 +40,51 @@
 public final class Cache implements ICache {
   private static final String CACHE_CFG_FILE = "ehcache.xml";
   private static final String CACHE_NAME = "pentaho-cde";
-  private Ehcache ehcache;
+  private static final String ENABLE_SHUTDOWN_HOOK_PROPERTY = "javax.cache.CacheManager.enableShutdownHook";
+  private javax.cache.Cache ehcache;
 
   public Cache( IContentAccessFactory contentAccessFactory ) throws InitializationException {
 
     CacheManager cacheManager;
     try {
-      cacheManager = CacheManager.create( contentAccessFactory.getPluginSystemReader( null )
-        .getFileInputStream( CACHE_CFG_FILE ) );
+      CachingProvider cachingProvider = Caching.getCachingProvider();
+      URI configUri = getClass().getClassLoader().getResource( CACHE_CFG_FILE ).toURI();
+      cacheManager = cachingProvider.getCacheManager( configUri, getClass().getClassLoader() );
+      InputStream inputStream = contentAccessFactory.getPluginSystemReader( null )
```

Comment:
Same as above!

Reply:
In the master, in order to read the configuration file we were using InputStream and provide the same to CacheManager. Here if there is any problem with the configuration file, it use to throw an error. Now this check is added to check if the configuration file exists and readable.

However, I have changed the logic to create a CacheManager. Removed usage of InputStream.